common/endian.h

```diff
@@ -623,6 +623,127 @@ inline void WRITE_BE_UINT24(void *ptr, uint32 value) {
 #define WRITE_UINT24(a,b) WRITE_BE_UINT24(a,b)
 #endif
 
+union SwapFloat {
+	float f;
+	uint32 u32;
+};
+
+STATIC_ASSERT(sizeof(float) == sizeof(uint32), Unexpected_size_of_float);
+
+inline float READ_LE_FLOAT32(const void *ptr) {
+	SwapFloat swap;
+	swap.u32 = READ_LE_UINT32(ptr);
+	return swap.f;
+}
+
+inline void WRITE_LE_FLOAT32(void *ptr, float value) {
+	SwapFloat swap;
+	swap.f = value;
+	WRITE_LE_FLOAT32(ptr, swap.u32);
```

Comment:
Shouldn't this be `UINT32`?

Reply:
Good catch - I've fixed this and added more tests for these functions.